lib/assets/data_table/main.js

```diff
@@ -0,0 +1,149 @@
+import * as Vue from "https://cdn.jsdelivr.net/npm/vue@3.2.26/dist/vue.esm-browser.prod.js";
+
+export function init(ctx, data) {
+  ctx.importCSS("main.css");
+  ctx.importCSS("https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600&display=swap");
+  ctx.importCSS("https://cdn.jsdelivr.net/npm/remixicon@2.5.0/fonts/remixicon.min.css");
+
+  const app = Vue.createApp({
```

Comment:
As for the JavaScript part, I started with re-rendering the content with `innerHTML` (which is fairly ok in this case, as we don't render that much), however this resets all elements, in particular horizontal scroll position. We could use morphdom to optimise the DOM patching, but then I thought we may as well use something a bit more friendly to make the overall implementation more robust. I went with Vue, since it's easy to drop into a single file and the template doesn't require any fancy transpilation. I think it came out pretty neat and makes for a good example too.